cmd/hegel/main.go

```diff
@@ -24,7 +30,183 @@ var (
 	customEndpoints string
 )
 
+const EnvNamePrefix = "HEGEL"
+
+type RootCommandOptions struct {
+	Facility       string `mapstructure:"facility"`
+	DataModel      string `mapstructure:"data-model"`
+	TrustedProxies string `mapstructure:"trusted-proxies"`
+
+	HTTPPort            int    `mapstructure:"http-port"`
+	HTTPCustomEndpoints string `mapstructure:"http-custom-endpoints"`
+
+	GrpcTLSCertPath string `mapstructure:"grpc-tls-cert"`
+	GrpcTLSKeyPath  string `mapstructure:"grpc-tls-key"`
+	GrpcUseTLS      bool   `mapstructure:"grpc-use-tls"`
+
+	KubeAPI    string `mapstructure:"kubernetes-api"`
+	Kubeconfig string `mapstructure:"kubeconfig"`
+}
+
+type RootCommand struct {
+	*cobra.Command
+	RootCommandOptions
+}
+
+func NewRootCommand() (*RootCommand, error) {
+	rootCmd := &RootCommand{
+		Command: &cobra.Command{
+			Use: os.Args[0],
+			Long: `
```

Comment:
Why define the help manually?

Reply:
To clarify, this doesn't define the complete help manually. It defines a long description that gets used with `-h`. The flags portion is still auto gen.

The purpose is to clarify deprecated vs current arguments and how things map. I felt it was appropriate to include the documentation until we remove things but I'm on the fence on whether its necessary.

Reply:
Resolving. Unresolve if further discussion needed.